### ISSUE TITLE
"process" not defined for babel-plugins (emotion)

### ISSUE THREAD

**Issue by sebald** [parcel-bundler/parcel#2237]
# 🐛 bug report

I was trying out `emotion` @ version `10.0.0-beta.7` with the babel-plugin.
With the default config:

```
{
  "plugins": ["emotion"]
}
```

the plugin seems to try to access `process` to determine if it should run in production or development mode (to add labels or minify), but fails since it can not access the variable.

```
Uncaught ReferenceError: process is not defined
    at Object.parcelRequire.src/ui/SomeComponent/SomeComponent.tsx.@emotion/styled-base
```

*Note: I am not 100% sure that this is an issue wie Parcel, could also be that the plugins does something it shouldn't do?*

## 🎛 Configuration (.babelrc, package.json, cli command)

```
// .babelrc
{
  "plugins": ["emotion"]
}
```

```
// package.json
{
  "main": "src/index.ts",
  "dependencies": {
    "@emotion/core": "^10.0.0-beta.7",
    "@emotion/styled": "^10.0.0-beta.7",
    "react": "^16.7.0-alpha.0",
    "react-dom": "^16.7.0-alpha.0"
  },
  "devDependencies": {
    "@types/react": "^16.4.18",
    "@types/react-dom": "^16.0.9",
    "parcel-bundler": "^1.10.3",
    "typescript": "^3.1.6"
  },
  "scripts": {
    "start": "parcel index.html"
  }
}
```

## 🤔 Expected Behavior

`emotion` can not access required ENV vars:

```
Uncaught ReferenceError: process is not defined
    at Object.parcelRequire.src/ui/SomeComponent/SomeComponent.tsx.@emotion/styled-base
```

## 😯 Current Behavior

`emotion` can not access the `process` var and throws.

## 🔦 Context

Trying the next version of `emotion` (v10).

## 💻 Code Sample

```tsx
SomeComponent.tsx

import * as React from 'react';
import styled from '@emotion/styled';

export const SomeComponent = styled.div`
  color: hotpink;
`;
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.10.3
| Node             | v10.8.0
| Yarn             | 1.9.4
| Operating System | MacOS Mojave

**Comment by XGHeaven** [comment]
Same error here

**Comment by alflennik** [comment]
There is a PR here that will resolve this issue (not merged yet) https://github.com/parcel-bundler/parcel/pull/2695

**Comment by mischnic** [comment]
> There is a PR here that will resolve this issue (not merged yet)

I think you misunderstood this [comment](https://github.com/parcel-bundler/parcel/issues/2470#issuecomment-468028575): that PR only fixes the JSX pragma overriding. This bug is caused by an unusual babel plugin implementation by emotion.

@DeMoorJasper  

> Not sure if moving it to after user transforms would have a big impact or not. But feel free to experiment with it and open a PR if it actually works better :)

https://github.com/parcel-bundler/parcel/blob/27f769a02ad9a54f8c1bac13961068cd34cdfd03/packages/core/parcel-bundler/src/assets/JSAsset.js#L153-L159

The issue is that the emotion babel plugin **injects** `process.env.NODE_ENV` into the generated code (but it seems to be the first plugin to do this, otherwise this would have been way reported earlier).
Reason: `await babel(this);` modifies the code include env expressions but `
ENV_RE.test(this.contents)` doesn't use the babel output but rather the original asset input. 
But doing the AST->code conversion twice seems very expensive (either explicitly or by letting `babel.transform` create it alongside the AST)
https://github.com/parcel-bundler/parcel/blob/27f769a02ad9a54f8c1bac13961068cd34cdfd03/packages/core/parcel-bundler/src/assets/JSAsset.js#L153-L159

**Comment by alflennik** [comment]
@devongovett I saw that [the PR](https://github.com/parcel-bundler/parcel/pull/2695) fixing this issue was closed because of work on Parcel 2. What are we expected to do right now regarding this issue?

**Comment by mischnic** [comment]
> I saw that the PR fixing this issue was closed because of work on Parcel 2. What are we expected to do right now regarding this issue?

(Again, we are really talking about #2470. This issue is not about the JSX pragma.)

Yes, the PR was considered too big of a potential breaking chance. Parcel 2 is going to use a .babelrc unmodified if provided (no strange overrides). I guess you'll have to use the workaround described in #2470.

**Comment by mischnic** [comment]
This will be fixed in Parcel 2 (#3334)
